### DISCUSSION TITLE
Azure FunctionsでPythonを使って外部ファイルを読み込む方法

### DISCUSSION THREAD

**Discussion by aono-tomo** [1ft-seabass/ms-azure-hackathon-league-2021#30] [Q&A] [answered]
OpenCVとpythonを使って目の位置を認識しようとしています。

``` python
face_cascade_path = '/usr/local/opt/haarcascade_frontalface_default.xml'
eye_cascade_path = '/usr/local/opt/haarcascade_eye.xml'

face_cascade = cv2.CascadeClassifier(face_cascade_path)
eye_cascade = cv2.CascadeClassifier(eye_cascade_path)

src_gray = cv2.cvtColor(src, cv2.COLOR_BGR2GRAY)

faces = face_cascade.detectMultiScale(src_gray)
```

上記のように、xmlファイルを読み込む必要があるんですがAzure Functionsで実行するとエラーが発生します。

```
error: (-215:Assertion failed) !empty() in function 'detectMultiScale'
```

何か解決する方法はありますでしょうか？

ちなみにGoogle colaboratoryでは動作しました。

**Comment by dzeyelid** [comment, marked as answer]
@bruno1911 その読み込みたい xml ファイルはどこに置いてあるファイルでしょうか？

文面から推測するに、お手元のマシン上にあるファイルを、Azure Functions で読もうとしているように見受けられますが、Functions の性質上、それはできません。（後述参照）

今回の場合は、xml ファイルを Azure Blob storage に置いて、SDK を使って読み込むとよいと思います。

- [azure-sdk-for-python/sdk/storage/azure-storage-blob at main · Azure/azure-sdk-for-python](https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/storage/azure-storage-blob)

なお、 Google colaboratory はブラウザ上で動作するサービスのようなので、Azure Functions とはサービス形態が異なります。Azure Functions は、分類するなればサーバーサイドのプログラムを稼働させるものなので、お手元のマシンとは接続できません。

**Reply by dzeyelid** [reply]
Zoom にて、上記はちょっとおこまりの内容と違うことを確認しました。

いったん、ファイルが読み込めていないのか、別の要因で `detectMultiScale` が失敗しているのか、もう少しデバッグしてもらうことになりました。
